web/src/components/storage/iscsi/InitiatorPresenter.jsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import { DropdownToggle, Skeleton } from "@patternfly/react-core";
+import { TableComposable, Thead, Tr, Th, Tbody, Td, ActionsColumn } from '@patternfly/react-table';
+
+import { Icon } from '~/components/layout';
+import { InitiatorForm } from "~/components/storage/iscsi";
+
+const RowActions = ({ actions, id, ...props }) => {
+  const actionsToggle = (props) => (
+    <DropdownToggle
+      id={id}
+      aria-label="Actions"
+      toggleIndicator={null}
+      isDisabled={props.isDisabled}
+      onToggle={props.onToggle}
+    >
+      <Icon name="more_vert" size="24" />
+    </DropdownToggle>
+  );
+
+  return (
+    <ActionsColumn
+      items={actions}
+      actionsToggle={actionsToggle}
+      {...props}
+    />
+  );
+};
+
+export default function InitiatorPresenter({ initiator, client }) {
+  const [data, setData] = useState();
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  useEffect(() => {
+    if (initiator !== undefined) setData({ ...initiator });
+  }, [setData, initiator]);
```

Comment:
Actually this is not needed and might be causing an extra render. Simply initialize data with the prop value.

```suggestion
export default function InitiatorPresenter({ initiator, client }) {
  const [data, setData] = useState(initiator);
  const [isFormOpen, setIsFormOpen] = useState(false);
```